java/memory/memory-core/src/test/java/org/apache/arrow/memory/CountingAllocationListener.java

```diff
@@ -58,6 +64,10 @@ public void onAllocation(long size) {
   @Override
   public boolean onFailedAllocation(long size, AllocationOutcome outcome) {
     if (expandOnFail) {
+      if (expandAlloc == null) {
```

Comment:
Could possibly use Preconditions.checkState for this (since it's been annotated such that CheckerFramework should understand it)